src/txmempool.h

```diff
@@ -14,6 +14,7 @@
 #include <utility>
 #include <vector>
 
+#include <chain.h>
```

Comment:
nit: could use the forward-declaration `class CChain;` instead to reduce header dependencies

Reply:
taken in #23649